### DISCUSSION TITLE
Passing variables to nested templates

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [hairyhenderson/gomplate#1731] [Q&A]
Hey, I have a template that includes a bunch of partials and I'd like to define variables to use just once. My template looks like this:

```
{{- if and (has .pkg "blueprint") (has .pkg.blueprint "helpDeskEmail") -}}
{{-   $helpDeskEmail := .pkg.blueprint.helpDeskEmail -}}
{{- else -}}
{{-   $helpDeskEmail := .global.helpDeskEmail -}}
{{- end -}}
{{- tmpl.Inline (include "git" "docs/CONTRIBUTING-header.md.tmpl") -}}
```

Then, in the included template, I'd like to be able to use `{{ $helpDeskEmail }}`. How can I accomplish this?

**Comment by hairyhenderson** [comment]
You'd need to pass that value in to the nested template's context. Your template would then need to refer to it as `.` or `.something`.

Also there's a scoping bug in your template currently - when you use `:=` that declares and defines the variable within that scope (the `if` block, in this case), so it won't be able to be referenced in the `tmpl.Inline` line. Instead, declare it with `:=`, then redefine it with `=`.

The final template would look something like:

```
{{ $helpDeskEmail := "" -}}
{{- if and (has .pkg "blueprint") (has .pkg.blueprint "helpDeskEmail") -}}
{{-   $helpDeskEmail = .pkg.blueprint.helpDeskEmail -}}
{{- else -}}
{{-   $helpDeskEmail = .global.helpDeskEmail -}}
{{- end -}}
{{- tmpl.Inline (include "git" "docs/CONTRIBUTING-header.md.tmpl") $helpDeskEmail -}}
```

Within the `docs/CONTRIBUTING-header.md.tmpl` you could read that context:

```
{{ $helpDeskEmail := . -}}
The helpdesk e-mail is: {{ $helpDeskEmail }}
```

_or, just:_

```
The helpdesk e-mail is: {{ . }}
```

If you want to pass multiple values through, use `dict`. See the examples in [the `tmpl.Inline` docs](https://docs.gomplate.ca/functions/tmpl/#tmplinline).

**Comment by ProfessorManhattan** [comment]
Thanks I'll take a look at the referenced docs

**Comment by ProfessorManhattan** [comment]
Hey @hairyhenderson -- do you know why the following code:

```
{{- if and (has .pkg "blueprint") (has .pkg.blueprint "helpDeskEmail") -}}
{{-   $helpDeskEmail := .pkg.blueprint.helpDeskEmail -}}
{{- else -}}
{{-   $helpDeskEmail := .global.helpDeskEmail -}}
{{- end -}}
{{- $dict := (coll.Dict "helpDeskEmail" $helpDeskEmail) -}}
{{- tmpl.Inline (include "git" "docs/CONTRIBUTING-header.md.tmpl") $dict -}}
```

Exits with the following error:

```
01:56:04 ERR  error="failed to render template common/docs/CONTRIBUTING.md.tmpl: template: common/docs/CONTRIBUTING.md.tmpl:2:10: executing \"common/docs/CONTRIBUTING.md.tmpl\" at <tmpl.Inline>: error calling Inline: template: <inline>:6: undefined variable \"$helpDeskEmail\""
```

**Reply by hairyhenderson** [reply]
I already explained that [above](https://github.com/hairyhenderson/gomplate/discussions/1731#discussioncomment-5877666):

> Also there's a scoping bug in your template currently - when you use `:=` that declares and defines the variable within that scope (the if block, in this case), so it won't be able to be referenced in the `tmpl.Inline` line. Instead, declare it with `:=`, then redefine it with `=`.
